fix: use 330 sat dust limit for P2TR/P2WPKH change outputs

Andezion opened the pull request:
(Fixes #8395)

## Problem

When creating transactions, change outputs between 330-546 sat were being treated as dust and absorbed into fees. This caused users to overpay transaction fees.

The issue was that change_amount() function used chainparams->dust_limit (546 sat) for all output types, but P2TR and P2WPKH outputs have a lower dust limit of 330 sat according to Bitcoin Core standards.

## Solution

Changed change_amount() in bitcoin/tx.c to use 330 sat dust limit for change outputs. This is correct because:
- Change outputs are P2TR on Bitcoin mainnet
- Change outputs are P2WPKH on Elements
- Both types have 330 sat dust limit (not 546 sat like legacy P2PKH/P2SH)

Now change outputs between 330-546 sat are properly created instead of being absorbed into fees.

## Testing

- Added new test test_p2tr_change_dust.py that verifies change outputs in 330-546 sat range are created
- Existing tests still pass: test_fundpsbt, test_withdraw_misc


> [!IMPORTANT]
>
> 26.04 FREEZE March 11th: Non-bugfix PRs not ready by this date will wait for 26.06.
>
> RC1 is scheduled on _March 23rd_
>
> The final release is scheduled for April 15th.


## Checklist
Before submitting the PR, ensure the following tasks are completed. If an item is not applicable to your PR, please mark it as checked:

- [ ] The changelog has been updated in the relevant commit(s) according to the [guidelines](https://docs.corelightning.org/docs/coding-style-guidelines#changelog-entries-in-commit-messages).
- [ ] Tests have been added or modified to reflect the changes.
- [ ] Documentation has been reviewed and updated as needed.
- [ ] Related issues have been listed and linked, including any that this PR closes.
- [ ] *Important* All PRs must consider how to reverse any persistent changes for `tools/lightning-downgrade`

Lagrang3 reviewed (changes requested):
First two commits should be removed.
Some changes afterwards seem unnecessary.
Also, the rest of the commits should be squashed into one, there is basically one
change and the rest is just formatting.

Andezion committed:
bitcoin/tx.c: use 330 sat dust limit for P2TR/P2WPKH change outputs Fixes #8395
Fix by checking is_elements: Elements keeps 546 sat, Bitcoin uses 330 sat.
Changelog-Fixed: Transactions now correctly create change outputs >= 330 sat for P2TR/P2WPKH instead of absorbing them as fees

madelinevibes commented:
@Andezion can you update per Eduardo's comments, resolve the conflicts and clear CI by the end of this week so we can add this to 26.06? Release candidate planned for Monday 11 May. 

If you can't we'll move it to 25.09

Lagrang3 commented on tests/test_p2tr_change_dust.py:
```diff
@@ -0,0 +1,54 @@
+    psbt = bitcoind.rpc.decodepsbt(result['psbt'])
+
+    change_outnum = result['change_outnum']
+    if 'tx' in psbt:
+        change_output = psbt['tx']['vout'][change_outnum]
+        change_amount_btc = float(change_output['value'])
+    else:
+        change_output = psbt['outputs'][change_outnum]
+        change_amount_btc = float(change_output['amount'])
```
Sorry for my ignorance. Is this condition here because `tx` could be absent from the `decodepsbt` in bitcoin core rpc?

Can't we just use one or the other?

Andezion replied:
This condition is present because the structure of the PSBT returned by Bitcoin Core's decodepsbt RPC can vary depending on the version and context. Sometimes, the decoded PSBT includes a full transaction under the tx key (with vout), and sometimes it only provides an outputs array. To ensure compatibility with both cases, the code checks for the presence of tx and uses the appropriate structure. So i guess we can't always use just one and the code needs to handle both formats for robustness :/ (if ive understood correctly)

Lagrang3 commented:
Just a minor comment up there and please remove the last commit "Merge branch master into fix-p2tr-dust-limit" which probably crept in by mistake.
The rest seems good to me.

Lagrang3 commented:
I've cleaned up the commits and rebased.
It should be good if tests pass.
